subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceFlakinessAnalyzer.groovy

```diff
@@ -20,26 +20,38 @@ import groovy.transform.CompileStatic
 import groovy.transform.TypeChecked
 import groovy.transform.TypeCheckingMode
 import org.gradle.ci.common.model.FlakyTest
+import org.gradle.ci.github.DefaultGitHubIssuesClient
 import org.gradle.ci.github.GitHubIssuesClient
+import org.gradle.ci.tagging.flaky.GitHubKnownIssuesProvider
 import org.gradle.ci.tagging.flaky.KnownFlakyTestProvider
 import org.gradle.performance.results.ScenarioBuildResultData.ExecutionData
 import org.kohsuke.github.GHIssue
 import org.kohsuke.github.GHIssueState
 
+import static org.apache.commons.lang3.StringUtils.isNoneBlank
 import static org.gradle.ci.github.GitHubIssuesClient.CI_TRACKED_FLAKINESS_LABEL
 import static org.gradle.ci.github.GitHubIssuesClient.FROM_BOT_PREFIX
 import static org.gradle.ci.github.GitHubIssuesClient.MESSAGE_PREFIX
 import static org.gradle.ci.github.GitHubIssuesClient.TEST_NAME_PREFIX
 
 @CompileStatic
-class FlakinessIssueReporter {
+class PerformanceFlakinessAnalyzer {
+
+    static PerformanceFlakinessAnalyzer create() {
+        GitHubIssuesClient gitHubIssuesClient = new DefaultGitHubIssuesClient(System.getProperty("githubToken"))
+        KnownFlakyTestProvider provider = new GitHubKnownIssuesProvider(gitHubIssuesClient)
+        return new PerformanceFlakinessAnalyzer(gitHubIssuesClient, provider)
+    }
+
     static final String GITHUB_FIX_IT_LABEL = "fix-it"
-    private final KnownFlakyTestProvider provider
+    static final String GITHUB_IN_PERFORMANCE_LABEL = "in:performance"
```

Comment:
Does this mean we should also only load flaky issues with the `in:performance` label for matching the existing performance tests?